dev-requirements.txt

```diff
@@ -1,37 +1,133 @@
 -i https://pypi.python.org/simple
+aiohttp==3.8.1
+aiosignal==1.3.1 ; python_version >= '3.7'
+ape-solidity==0.6.3
+appnope==0.1.3 ; sys_platform == 'darwin'
+asttokens==2.2.1
+async-timeout==4.0.2 ; python_version >= '3.6'
 attrs==23.1.0 ; python_version >= '3.7'
+backcall==0.2.0
+base58==1.0.3
+bitarray==2.7.3
+cached-property==1.5.2
 certifi==2022.12.7 ; python_version >= '3.6'
+cffi==1.15.1
 cfgv==3.3.1 ; python_full_version >= '3.6.1'
 charset-normalizer==2.1.1 ; python_full_version >= '3.6.0'
+click==8.1.3
+commonmark==0.9.1
 coverage==6.5.0
+cryptography==40.0.2
+cytoolz==0.12.1 ; implementation_name == 'cpython'
+dataclassy==0.11.1 ; python_version >= '3.6'
 decorator==5.1.1 ; python_version >= '3.5'
+deprecated==1.2.13 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
 distlib==0.3.6
+eip712==0.2.1 ; python_version >= '3.8' and python_version < '4'
+eth-abi==4.0.0 ; python_version >= '3.7' and python_version < '4'
+eth-account==0.8.0 ; python_version >= '3.6' and python_version < '4'
+eth-ape==0.6.8
+eth-bloom==2.0.0 ; python_version >= '3.7' and python_version < '4'
+eth-hash==0.5.1 ; python_version >= '3.7' and python_version < '4'
+eth-keyfile==0.6.1
+eth-keys==0.4.0
+eth-rlp==0.3.0 ; python_version >= '3.7' and python_version < '4'
+eth-tester[py-evm]==0.8.0b3
+eth-typing==3.3.0 ; python_full_version >= '3.7.2' and python_version < '4'
+eth-utils==2.1.0
+ethpm-types==0.4.4 ; python_version < '3.11' and python_version >= '3.8'
+evm-trace==0.1.0a18 ; python_version >= '3.8' and python_version < '4'
 exceptiongroup==1.1.1 ; python_version < '3.11'
+executing==1.2.0
 filelock==3.12.0 ; python_version >= '3.7'
+frozenlist==1.3.3 ; python_version >= '3.7'
 greenlet==2.0.2 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'
-hypothesis==6.72.1
-identify==2.5.22 ; python_version >= '3.7'
+hexbytes==0.3.0 ; python_version >= '3.7' and python_version < '4'
+hypothesis==6.74.1
+identify==2.5.23 ; python_version >= '3.7'
 idna==3.4 ; python_version >= '3.5'
+ijson==3.2.0.post0
+importlib-metadata==6.6.0 ; python_version >= '3.7'
 iniconfig==2.0.0 ; python_version >= '3.7'
+ipython==8.13.0 ; python_version >= '3.8'
```

Comment:
I'm wondering how this dependency was resolved.  Did you build this from python 3.8.x?


```
...
ImportError: 
IPython 8.13+ supports Python 3.9 and above, following NEP 29.
IPython 8.0-8.12 supports Python 3.8 and above, following NEP 29.
```

Reply:
> Did you relock from python 3.8.x?
I have been using python 3.8.16